microsoft-edge/devtools-guide-chromium/memory-problems/dom-leaks-memory-tool-detached-elements.md

```diff
@@ -0,0 +1,139 @@
+
+
+   <!-- ------------------------------ -->
+   **Change to a different instance of the Room class, so elements become detached:**
+
+1. In the demo webpage, click the **Room 2** button, which corresponds to another instance of the `Room` class.
+
+   In the webpage, the messages disappear:
+
+   ![Room 2 initial view](./dom-leaks-memory-tool-detached-elements-images/room-2-initial-view.png)
+
+   The messages that were generated for the Room 1 instance of the **Room** class (`<div class="message">` elements) are no longer attached to the DOM, but they're still referenced by the Room 1 instance of the **Room** class.  They are detached elements, which can cause memory leaks, unless they are going to be used again by the webpage.
+
+
+   <!-- ------------------------------ -->
+   **Get the list of detached elements:**
+
+1. In DevTools, in the **Memory** tool, select the **Detached elements** option button, and then click the **Start** button.
+
+   The list of detached nodes is displayed in the **Detached nodes** column of the generated **Detached elements** profile:
+
+   ![List of detached nodes in the Detached nodes column of the generated Detached elements profile](./dom-leaks-memory-tool-detached-elements-images/detached-dom-nodes.png)
+
+   The detached elements are displayed, as DOM nodes like in the DOM tree in the **Elements** tool.  In the **Profiles** list, in the **Detached elements** section, the profile is listed.
+
+1. Expand a DOM node to display its child elements (DOM nodes):
+
+   ![An expanded detached DOM node](./dom-leaks-memory-tool-detached-elements-images/detached-dom-node-expanded.png)
+
+   These detached elements (DOM nodes) are memory leaks, if they aren't going to be reused by the application.
+
+
+If you want to return to the list of option buttons for profiling types, in the upper left of the **Memory** tool, click **Profiles** (![the Profiles icon](./dom-leaks-memory-tool-detached-elements-images/profiles-icon.png)).
+
+For additional ways to assess memory leaks, see [Tools for investigating detached elements](./index.md#tools-for-investigating-detached-elements) in _Fix memory problems_.
+
+
+<!-- ====================================================================== -->
+## About DOM leaks due to detached elements
+
+To increase the performance of your webpage, find the elements that are detached from the DOM tree and which you didn't expect to still be referenced by JavaScript code.  Find the detached elements that the browser can't garbage-collect because your code still references them, and then release the JavaScript code references to the detached elements.
+<!-- copied from dom-leaks.md -->
+
+
+<!-- copied from dom-leaks.md, & trimmed to omit DE tool's UI controls -->
+<!-- ====================================================================== -->
+## About detached elements and memory leaks
```

Comment:
result:

**About detached elements and memory leaks**

Detached elements aren't always an indication of a memory leak, and memory leaks aren't always caused by detached elements.  Memory leaks depend on the context of your application.

A memory leak can occur in your application when an element is no longer attached to the Document Object Model (DOM) tree, but is still referenced by some JavaScript running on the webpage. These elements are called *detached elements*.  For the browser to garbage-collect (GC) the detached element, the element must not be referenced from the DOM tree or from JavaScript code.

Memory issues affect webpage performance, including memory leaks, memory bloat, and frequent garbage collections.  Symptoms for your users include:

*  The performance of a webpage gets progressively worse over time.
*  The performance of a webpage is consistently bad.
*  The performance of a webpage is delayed or appears to pause frequently.

To increase the performance of your webpage, find the elements that are detached from the DOM tree and which you didn't expect to still be referenced by JavaScript code.  Find the detached elements that the browser can't garbage-collect because your code still references them, and then release the JavaScript code references to the detached elements.